stream-chat-android-state/src/main/java/io/getstream/chat/android/offline/plugin/logic/querychannels/internal/QueryChannelsLogic.kt

```diff
@@ -151,16 +150,32 @@ internal class QueryChannelsLogic(
     }
 
     suspend fun onQueryChannelsResult(result: Result<List<Channel>>, request: QueryChannelsRequest) {
+        logger.d { "[onQueryChannelsResult] result.isSuccess: ${result.isSuccess}, request: $request" }
         onOnlineQueryResult(result, request, repos, globalState)
         if (result.isSuccess) {
-            updateOnlineChannels(result.data(), request.isFirstPage)
+            updateOnlineChannels(request, result.data())
         }
         val loading = loadingForCurrentRequest()
         loading.value = false
     }
 
-    internal suspend fun runQueryOnline(request: QueryChannelsRequest): Result<List<Channel>> {
-        return client.queryChannelsInternal(request).await()
+    /**
+     * Runs [QueryChannelsRequest] which is querying the first page.
+     */
+    internal suspend fun queryFirstPage(): Result<List<Channel>> {
+        logger.d { "[queryFirstPage] no args" }
+        val state = mutableState
+        val request = QueryChannelsRequest(
```

Comment:
@kanat what in a situation when someone uses different limits/offsets?

Reply:
This logic was here before behind the method `runQueryOnline`. I just move some parts from SyncManager to avoid ping-pong logic between `SyncManager` and `QueryChannelsLogic`.

This PR should fix the major issue we have.
The rest of the changes are related to unnecessary QueryChannels requests I'm planning to take during the sync refactoring.

Didn't want to cover a huge topic in this PR